### PR TITLE
Upgrade d3 to v7.4

### PR DESCRIPTION
**Pull request by shifucun**
Also

* Upgraded jest versions
* Removed babel libs since they are not used.

### CODE REVIEW

File: static/js/chart/draw_d3_map.ts [hunk 1]
```diff
@@ -342,9 +342,7 @@ export function getProjection(
       projection = d3.geoEquirectangular().rotate([-100, 0]).precision(0.1);
       break;
     default:
-      projection = isUSAPlace
-        ? geo.geoAlbersUsaTerritories()
-        : d3.geoEquirectangular();
+      projection = isUSAPlace ? d3.geoAlbersUsa() : d3.geoEquirectangular();
```
Comment (shifucun) [inline static/js/chart/draw_d3_map.ts:345]: @chejennifer which page should I verify for this projection change?

Comment (chejennifer) [reply static/js/chart/draw_d3_map.ts:345]: you can check any place that has US map charts - probably want to check US and a US state and US territory so maybe: 

/place/country/USA
/place/geoId/06
/place/geoId/72

File: packages/client/package.json [hunk 2]
```diff
@@ -47,14 +44,13 @@
     "@types/react-dom": "^18.2.17",
     "@types/react-syntax-highlighter": "^15.5.11",
     "@types/turf": "^3.5.32",
-    "babel-jest": "^29.7.0",
     "d3-scale": "^4.0.2",
     "jest": "^29.7.0",
```
Comment (shifucun) [inline packages/client/package.json:48]: @dwnoble This file and static/package.json has some common deps. I wonder if some more can be removed from here?

File: static/js/chart/draw_d3_map.ts (hunk 1 above)
Comment (shifucun) [reply static/js/chart/draw_d3_map.ts]: Reverted as the lib does not block the version update

File: static/js/biomedical/protein/chart.ts [hunk 3]
```diff
@@ -968,15 +969,16 @@ export function drawVarGeneAssocChart(
     .attr("cy", (d) => y(d.id))
     .attr("r", "6")
     .attr("id", (d, i) => circleIdFunc(i))
-    .style("fill", (d) => ERROR_BAR_VAR_COLOR[d.name])
+    .style("fill", (d: VarGeneDataPoint) => ERROR_BAR_VAR_COLOR[d.name])
     // variant redirect
-    .on("click", (d) => {
+    .on("click", (d: VarGeneDataPoint) => {
```
Comment (chejennifer) [inline static/js/biomedical/protein/chart.ts:983]: I see you updated a lot of the other `.on(<event>, () => {})` calls by adding an extra `event` param in the callback function, do you need to do that here as well?

File: static/js/tools/timeline/__snapshots__/page.test.tsx.snap [hunk 4]
```diff
@@ -1,169 +1,169 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`Single place and single stat var 1`] = `
-"<div class=\\"chart-container dc-async-element-holder\\">
-  <div class=\\"card\\">
-    <div class=\\"statVarChipRegion\\">
-      <div class=\\"chip\\"><span class=\\"chip-text chip-clickable-text\\">Age</span><button class=\\"chip-action\\"><i class=\\"material-icons\\">cancel</i></button></div>
+"<div class="chart-container dc-async-element-holder">
```
Comment (chejennifer) [inline static/js/tools/timeline/__snapshots__/page.test.tsx.snap:4]: is this updated because of jest updates?

**Review by chejennifer — APPROVED** [review]
nice!

File: static/js/tools/timeline/__snapshots__/page.test.tsx.snap (hunk 4 above)
Comment (shifucun) [reply static/js/tools/timeline/__snapshots__/page.test.tsx.snap:4]: Yes!

File: static/js/biomedical/protein/chart.ts (hunk 3 above)
Comment (shifucun) [reply static/js/biomedical/protein/chart.ts:983]: Good catch. Fixed here and a few other places.

**Pull request merged**